dolphinscheduler-registry/dolphinscheduler-registry-api/src/main/java/org/apache/dolphinscheduler/registry/api/ha/AbstractHAServer.java

```diff
@@ -78,21 +83,30 @@ public boolean isActive() {
     @Override
     public boolean participateElection() {
         final String electionLock = selectorPath + "-lock";
-        try {
-            if (registry.acquireLock(electionLock)) {
-                if (!registry.exists(selectorPath)) {
-                    registry.put(selectorPath, serverIdentify, true);
-                    return true;
+        // If meet exception during participate election, will retry.
+        // This can avoid the situation that the server is not elected as leader due to network jitter.
+        for (int i = 0; i < DEFAULT_MAX_RETRY_TIMES; i++) {
+            try {
+                try {
+                    if (registry.acquireLock(electionLock)) {
+                        if (!registry.exists(selectorPath)) {
+                            registry.put(selectorPath, serverIdentify, true);
+                            return true;
+                        }
+                        return serverIdentify.equals(registry.get(selectorPath));
+                    }
+                    return false;
+                } finally {
+                    registry.releaseLock(electionLock);
                 }
```

Comment:
As the issue avoid to use lock, but it's actually still using lock, what's the difference?

Reply:
The lock is only used when cluster changed, after selected an active server, the lock will be released, so the lock will only hold for a short time. This is only used for election, we use the lock here to make sure all registry plugins can implement the election method in an easy way.

Reply:
If we add another coordintor, will all distributed services be executed on this one active master? It seems the Master HA but not some logic coordinator HA, I'm not sure if this adjustment will cause one active master to be busy while others are idle.
```
public class MasterCoordinator extends AbstractHAServer {
    private final ITaskGroupCoordinator taskGroupCoordinator;
}
```

Reply:
Yes, the coordinator master shouldn't do a lot of business work, it would be great if it only deals with coordination work. At our usage case, the coordinator master will only coordinate task group, deal with serial wait this will not affect even though the master is busy, these work will not be affect.

BTW, after fixing a lot of memory leak/thread leak bug, I haven't found a master status change to BUSY in our cluster, currently, there is basically no heavy workload in the masters, most case is db busy.